test/functional/cursor_tests.js

```diff
@@ -4423,4 +4423,79 @@ describe('Cursor', function() {
       });
     });
   });
+
+  function testTransformStream(client, configuration, dbName, transformFunc, expectedSet, done) {
```

Comment:
1. I really like you making this single utility function.
2. You are using `dbName` here as a collection name. You should rename it.
3. This is a lot of parameters for a function call. I'd replace these with just two:

```js
function testTransformStream(config, done) {
```

And you can stick everything you need in `config`